src/components/SQFormDialogStepper/SQFormDialogStepper.js

```diff
@@ -103,41 +106,21 @@ export function SQFormDialogStepper({
     return activeStep === steps.length - 1;
   }, [activeStep, steps]);
 
-  // Our last step doesn't get marked complete
-  const isAllStepsCompleted = () => {
-    return Object.keys(completed).length === steps.length - 1;
-  };
-
   const handleNext = () => {
-    const newActiveStep =
-      isLastStep && !isAllStepsCompleted
-        ? // It's the last step, but not all steps have been completed,
-          // find the first step that has been completed
-          steps.findIndex((step, i) => !(i in completed))
-        : activeStep + 1;
-    setActiveStep(newActiveStep);
-    handleComplete();
+    setActiveStep(activeStep + 1);
+    setCompleted([...completed, activeStep]);
   };
 
   const handleStep = step => () => {
-    const nextStep = step.toString();
-    const prevStep = (step - 1).toString();
-    const completedKeys = Object.keys(completed);
-    if ([nextStep, prevStep].some(step => completedKeys.includes(step))) {
+    const nextStep = step;
+    if ([nextStep].some(step => completed.includes(step))) {
       setActiveStep(step);
     }
   };
 
-  const handleComplete = () => {
```

Comment:
This has been replaced by line 111 and the data is now in an array which makes things much simpler.